src/audio/node.rs

```diff
@@ -46,6 +46,12 @@ pub trait AudioNodeEngine: Send {
     fn process(&mut self, inputs: Chunk, info: &BlockInfo) -> Chunk;
 
     fn message(&mut self, _: AudioNodeMessage, _sample_rate: f32) {}
+
+    fn input_count(&self) -> u32 { 1 }
+    fn output_count(&self) -> u32 { 1 }
+
+    /// If we're the destination node, extract the contained data
+    fn destination_data(&mut self) -> Option<Chunk> { None }
```

Comment:
Could we reuse `process` for this? With an extra flag in `BlockInfo` probably.

Reply:
How? BlockInfo is immutable.

I don't want to do the model of the destination node pushing directly to the sink because the spec represents this as a separate "record the input for future retrieval" step. But we can still do that for now if you think it's better.

Reply:
Ok. I guess this is fine then.

I reverted 6131d39e6bdc9e300231c768a0b389e097c68fd9